package/MDAnalysis/lib/log.py

```diff
@@ -296,11 +320,12 @@ def echo(self, step, **kwargs):
             return
         self.update(step, **kwargs)
         format = self.format
+        newline = not self.dynamic
         if self.step == self.numsteps:
-            if self.last_newline:
-                format = self.format[:-1] + self.last_newline
+            newline = True
         elif self.numouts % self.interval == 0:
             pass
         else:
             return
-        echo(format % vars(self))
+        echo(format.format(**vars(self)),
```

Comment:
the new PCA analysis is using the extra format. I'm ok to use the format syntax since it's nicer. But our own code should be changed and it should be mentioned in the changelog as it is a mayor change.

Solution 4
- over a keyword for the old-syntax that is active by default and add a deprecation warning when the old format is used. Then we can drop the old format in about 1-2 releases down the road and we don't break user code without a phase to adjust for them.